File: interest-rate-lambda-demo/src/main/java/de/senacor/innolab/graalvm/demo/lambda/model/CreditDetails.java
package de.senacor.innolab.graalvm.demo.lambda.model;

import io.quarkus.runtime.annotations.RegisterForReflection;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@RegisterForReflection
public class CreditDetails {
    private BigDecimal amount;
    private LocalDate start;
    private LocalDate end;
}

File: src/main/kotlin/com/senacor/innolab/graalvm/credit/details/model/CreditDetails.kt
package com.senacor.innolab.graalvm.credit.details.model

import io.micronaut.core.annotation.Introspected
import com.fasterxml.jackson.annotation.JsonFormat
import java.math.BigDecimal
import java.time.LocalDate

@Introspected
class CreditDetails(

    var id: String,

    var amount: BigDecimal,

    @JsonFormat(pattern = "yyyy-MM-dd")
    var start: LocalDate,

    @JsonFormat(pattern = "yyyy-MM-dd")
    var end: LocalDate,

    var interestRate: BigDecimal
) {

    companion object {
        fun of(id: String, amount: BigDecimal, startDate: String, endDate: String, interestRate: BigDecimal) =
            CreditDetails(
                id,
                amount,
                LocalDate.parse(startDate),
                LocalDate.parse(endDate),
                interestRate
            )
    }
}